### PR TITLE
Remove various Python 2 version_info checks

### PR DESCRIPTION
**Pull request by Avasam**
https://github.com/pyasn1/pyasn1/commit/81ce8d0d5f1c6ae4662cdc1fe141b7ea0cd28e00 removed support for Python 2, but there's still tons of leftover python2 dead code.

Here's a cleanup of Python 2 version_info checks, except for  compat.integer (https://github.com/pyasn1/pyasn1/pull/61) and compat.octets (https://github.com/pyasn1/pyasn1/pull/60)

Hiding whitespace changes make reviewing these changes much easier.

### CODE REVIEW

**Review by mshafer-NI — COMMENTED** [review]
Reading through from a Python perspective this looks good.

**Review by droideck — APPROVED** [review]
Besides the minor issues, looks good! Thank you!

File: pyasn1/type/base.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ def __repr__(self):
         return '<%s>' % representation
 
     def __eq__(self, other):
-        return self is other and True or self._value == other
+        return self is other or self._value == other
```
[the remark this answers is not in the document]
Comment (Avasam) [reply pyasn1/type/base.py:293]: Your preference. Either work 🙂
```suggestion
        if self is other:
            return True
        return self._value == other
```

File: pyasn1/type/constraint.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def __repr__(self):
         return '<%s>' % representation
 
     def __eq__(self, other):
-        return self is other and True or self._values == other
+        return self is other or self._values == other
```
[the remark this answers is not in the document]
Comment (Avasam) [reply pyasn1/type/constraint.py:49]:
```suggestion
        if self is other:
            return True
        return self._value == other
```

**Pull request merged**